network/eos/eos_eapi.py

```diff
@@ -35,259 +35,335 @@
+
+def set_protocol_https(module, commands):
+    port = module.params['https_port']
+    if not 0 <= port <= 65535:
+        module.fail_json(msg='https_port must be between 1 and 65535')
+    elif module.params['https'] is True:
+        commands.append('protocol https port %s' % port)
+    elif module.params['https'] is False:
+        commands.append('no protocol https')
+
+def set_local_http(module, commands):
+    port = module.params['local_http_port']
+    if not 0 <= port <= 65535:
+        module.fail_json(msg='local_http_port must be between 1 and 65535')
+    elif module.params['local_http'] is True:
+        commands.append('protocol http localhost port %s' % port)
+    elif module.params['local_http'] is False:
+        commands.append('no protocol http localhost port 8080')
+
+def set_socket(module, commands):
+    if module.params['socket'] is True:
         commands.append('protocol unix-socket')
-        result['changed'] = True
-
-    if len(commands) > 1:
-        # something requires change
-        execute_commands(module, commands)
-        result['commands'] = commands
+    elif module.params['socket'] is False:
+        commands.append('no protocol unix-socket')
 
-    return result
+def set_vrf(module, commands):
+    vrf = module.params['vrf']
+    if vrf != 'default':
+        resp = module.cli(['show vrf'])
+        if vrf not in resp[0]:
+            module.fail_json(msg="vrf '%s' is not configured" % vrf)
+    commands.append('vrf %s' % vrf)
+
+def set_qos(module, commands):
+    if not 0 <= module.params['qos'] <= 63:
+        module.fail_json(msg='qos must be between 0 and 63')
+    commands.append('qos dscp %s' % module.params['qos'])
+
+def get_config(module):
+    contents = module.params['config']
+    if not contents:
+        cmd = 'show running-config all section management api http-commands'
+        contents = module.cli([cmd])
+    config = NetworkConfig(indent=3, contents=contents[0])
+    return config
+
+def load_config(module, commands, result):
+    session = 'ansible_%s' % int(time.time())
+    commit = not module.check_mode
+
+    diff = module.config.load_config(commands, session=session, commit=commit)
+
+    # once the configuration is done, remove the config session and
+    # remove the session name from the result
+    module.cli(['no configure session %s' % session])
+
+    result['diff'] = dict(prepared=diff)
+    result['changed'] = diff is not None
+
+def load(module, commands, result):
+    candidate = NetworkConfig(indent=3)
+    candidate.add(commands, parents=['management api http-commands'])
+
+    config = get_config(module)
+    configobjs = candidate.difference(config)
+
+    if configobjs:
+        commands = dumps(configobjs, 'commands').split('\n')
+        result['updates'] = commands
+        load_config(module, commands, result)
+
+def clean_result(result):
+    # strip out any keys that have two leading and two trailing
+    # underscore characters
```

Comment:
keys in result that take on the format __keyname__ are internal to the module processing so I strip them out before returning the result dict